src/ui_kit/src/file/mod.rs

```diff
@@ -19,19 +26,130 @@ pub fn File<'a>(cx: Scope<'a, Props>) -> Element<'a> {
         State::Secondary => "secondary",
     };
 
-    let file_name = format_file_name_to_show(cx);
+    let file_id = cx.props.id.clone();
+
+    let file_name_fmt = format_file_name_to_show(cx.props.name.clone(), cx.props.kind.clone());
+
+    let file_name_formatted_state = use_state(&cx, || file_name_fmt);
+
+    let file_name_complete_ref = use_ref(&cx, || cx.props.name.clone());
 
-    let file_size = format_file_size(cx.props.size);
+    let file_size = format_file_size(cx.props.size);    
+
+    let show_edit_name_script = include_str!("./show_edit_name.js").replace("file_id", &file_id);
+    let hide_edit_name_script = include_str!("./hide_edit_name.js").replace("file_id", &file_id);
 
     cx.render(rsx! {
         div {
-            class: "folder {class}",
+            class: "item file",
+            id: "{file_id}-file",
+                ContextMenu {
+                    parent: format!("{}-file", file_id.clone()),
+                    items: cx.render(
+                        rsx! {
+                                ContextItem {
+                                    icon: Shape::PencilSquare,
+                                    onpressed: move |_| {
+                                        // Use js script to show edit file name element
+                                        use_eval(&cx)(&show_edit_name_script);
```

Comment:
Something to becareful about when doing it this way as replacing a section of the script while loading it in (as you are doing above) and evaluating it here could open possibly of an xss (though info wont be impacted but could still open holes if one is not careful with what its being replaced with, etc. Did leave a suggestion just adding a comment for it. Also using this function could panic if the js is invalid so something to becareful of too. 

```suggestion
                                        //TODO: Investigate in a way to replace use_eval in the future
                                        use_eval(&cx)(&show_edit_name_script);
```

Reply:
Thanks Darius, I added TODO, something important to care. I used use_eval as recommendation from dioxus lab devs, maybe in master, should have better ways to run a js script.

I tested a invalid js, and the app didn't have panic, just not worked. 

https://user-images.githubusercontent.com/63157656/204306638-7b350a59-8dd7-4f2a-ae60-fe30a1856e34.mov